ocaml/xapi/xapi_observer.ml

```diff
@@ -152,7 +163,8 @@ let assert_valid_hosts ~__context hosts =
 let assert_valid_components components =
   List.iter
     (fun component ->
-      if not (List.mem component supported_components) then
+      if not (List.mem component (List.map Component.to_string Component.all))
+      then
         raise
           Api_errors.(Server_error (invalid_value, ["component"; component]))
     )
```

Comment:
We can exploit exceptions here:
```
      try List.find (Component.of_string component) Component.all
      with _ ->
          raise
            Api_errors.(Server_error (invalid_value, ["component"; component]))
```

Reply:
or maybe just a `List.iter` to let the `of_string` raise the exception

Reply:
I don't understand, there's only one string that needs to be verified, so why would List.iter be useful here?

Reply:
ah I mean remove the `List.find` and let `of_string` raise the exception. Look at my new commits and see what you think

Reply:
ah, much better